src/plans/stepsize.jl

```diff
@@ -1687,3 +1687,193 @@ end
+
+Compute the geometric curvature function ``ζ_κ(d)`` for sectional curvature bound ``κ``.
+
+```math
+ζ_κ(d) = \\begin{cases}
+    \\frac{\\sqrt{|κ|} d}{\\tanh(\\sqrt{|κ|} d)} & \\text{if } κ < 0 \\\\
+    1 & \\text{if } κ ≥ 0
+\\end{cases}
+```
+"""
+function geometric_curvature_function(κ::Real, d::Real)
+    if κ < 0 && d > 0
+        sqrt_abs_κ = sqrt(abs(κ))
+        arg = sqrt_abs_κ * d
+        # Avoid numerical issues for small arguments
+        if arg < 1.0e-8
+            return 1.0 + arg^2 / 3.0  # Taylor expansion
+        else
+            return arg / tanh(arg)
+        end
+    else
+        return 1.0
+    end
+end
+
+function (rdog::DistanceOverGradientsStepsize)(
+        mp::AbstractManoptProblem,
+        s::AbstractManoptSolverState,
+        i,
+        args...;
+        gradient = nothing,
+        kwargs...,
+    )
+    M = get_manifold(mp)
+    p = get_iterate(s)
+    grad = isnothing(gradient) ? get_gradient(mp, p) : gradient
+
+    # Compute gradient norm
+    grad_norm_sq = norm(M, p, grad)^2
+
+    if i == 0
+        # Initialize on first call
+        rdog.gradient_sum = grad_norm_sq
+        rdog.initial_point = copy(M, p)
+        rdog.max_distance = rdog.initial_distance
+
+        # Initial stepsize
+        if rdog.use_curvature
+            ζ = geometric_curvature_function(
+                rdog.sectional_curvature_bound, rdog.max_distance
+            )
+            stepsize = rdog.initial_distance / (sqrt(ζ) * sqrt(max(grad_norm_sq, eps())))
+        else
+            stepsize = rdog.initial_distance / sqrt(max(grad_norm_sq, eps()))
+        end
+    else
+        # Update gradient sum
+        rdog.gradient_sum += grad_norm_sq
+
+        # Update max distance
+        current_distance = distance(M, rdog.initial_point, p)
+        rdog.max_distance = max(rdog.max_distance, current_distance)
+
+        # Compute stepsize
+        if rdog.use_curvature
+            ζ = geometric_curvature_function(
+                rdog.sectional_curvature_bound, rdog.max_distance
+            )
+            stepsize = rdog.max_distance / (sqrt(ζ) * sqrt(rdog.gradient_sum))
+        else
+            stepsize = rdog.max_distance / sqrt(rdog.gradient_sum)
+        end
+    end
+
+    rdog.last_stepsize = stepsize
+    return stepsize
+end
+
+get_initial_stepsize(rdog::DistanceOverGradientsStepsize) = rdog.last_stepsize
+get_last_stepsize(rdog::DistanceOverGradientsStepsize) = rdog.last_stepsize
+
+function show(io::IO, rdog::DistanceOverGradientsStepsize)
+    s = """
+    DistanceOverGradients(;
+      initial_distance=$(rdog.initial_distance),
+      use_curvature=$(rdog.use_curvature),
+      sectional_curvature_bound=$(rdog.sectional_curvature_bound)
+    )
+
+    Current state:
+      max_distance = $(rdog.max_distance)
+      gradient_sum = $(rdog.gradient_sum)
+      last_stepsize = $(rdog.last_stepsize)
+    """
+    return print(io, s)
+end
+
+"""
+    DistanceOverGradients(; kwargs...)
```

Comment:
The `docs/src/stepsize.md` has a block (the third code block) documenting all available step sizes – can you please add this there as well? 

Currently it is automatically added a bit later in the file - but it should be listed in the available step sizes.